src/test/scala/org/broadinstitute/dsde/workbench/sam/service/UserServiceSpec.scala

```diff
@@ -37,167 +53,318 @@ class UserServiceSpec
     with ScalaFutures
     with OptionValues {
 
-  override implicit val patienceConfig = PatienceConfig(timeout = scaled(5.seconds))
+  override implicit val patienceConfig: PatienceConfig = PatienceConfig(timeout = scaled(5.seconds))
   implicit override val generatorDrivenConfig: PropertyCheckConfiguration = PropertyCheckConfiguration(minSuccessful = 100)
 
   val defaultUser = genWorkbenchUserBoth.sample.get
+  val enabledDefaultUserStatus = UserStatusBuilder(defaultUser)
+    .withAllUsersGroup(true)
+    .withLdap(true)
+    .withGoogle(true)
+    .withTosAccepted(true)
+    .withAdminEnabled(true)
+    .build
 
   lazy val petServiceAccountConfig = TestSupport.appConfig.googleConfig.get.petServiceAccountConfig
-  lazy val dirDAO: DirectoryDAO = new PostgresDirectoryDAO(TestSupport.dbRef, TestSupport.dbRef)
 
   var service: UserService = _
-  var tos: TosService = _
-  var serviceTosEnabled: UserService = _
-  var tosServiceEnabled: TosService = _
   var googleExtensions: GoogleExtensions = _
+  var dirDAO: DirectoryDAO = _
+  var mockTosService: TosService = _
   val blockedDomain = "blocked.domain.com"
+  val enabledUser: SamUser = defaultUser.copy(enabled = true)
+  val disabledUser: SamUser = defaultUser.copy(enabled = false)
+  val allUsersGroup: BasicWorkbenchGroup = BasicWorkbenchGroup(WorkbenchGroupName("All_Users"), Set(), WorkbenchEmail("all_users@fake.com"))
 
   before {
-    clearDatabase()
+    dirDAO = mock[DirectoryDAO](RETURNS_SMART_NULLS)
+
+    when(dirDAO.loadUser(defaultUser.id, samRequestContext)).thenReturn(IO(Some(enabledUser)))
+    when(dirDAO.loadSubjectFromGoogleSubjectId(defaultUser.googleSubjectId.get, samRequestContext)).thenReturn(IO(None))
+    when(dirDAO.loadSubjectFromEmail(defaultUser.email, samRequestContext)).thenReturn(IO(Option(defaultUser.id)))
+    when(dirDAO.createUser(defaultUser, samRequestContext)).thenReturn(IO(disabledUser))
+    when(dirDAO.deleteUser(defaultUser.id, samRequestContext)).thenReturn(IO(()))
+    when(dirDAO.enableIdentity(defaultUser.id, samRequestContext)).thenReturn(IO(()))
+    when(dirDAO.disableIdentity(defaultUser.id, samRequestContext)).thenReturn(IO(()))
+    when(dirDAO.addGroupMember(allUsersGroup.id, defaultUser.id, samRequestContext)).thenReturn(IO(true))
+    when(dirDAO.removeGroupMember(allUsersGroup.id, defaultUser.id, samRequestContext)).thenReturn(IO(true))
+    when(dirDAO.isGroupMember(allUsersGroup.id, defaultUser.id, samRequestContext)).thenReturn(IO(true))
+    when(dirDAO.isEnabled(defaultUser.id, samRequestContext)).thenReturn(IO(true))
+    when(dirDAO.listUserDirectMemberships(defaultUser.id, samRequestContext)).thenReturn(IO(LazyList(allUsersGroup.id)))
+    when(dirDAO.setGoogleSubjectId(defaultUser.id, defaultUser.googleSubjectId.get, samRequestContext)).thenReturn(IO(()))
+    when(dirDAO.setUserAzureB2CId(defaultUser.id, defaultUser.azureB2CId.get, samRequestContext)).thenReturn(IO(()))
```

Comment:
maybe this block could be in a well-named fn just so you can understand what it is doing at a glance. I think that could be a followup pr though.